src/core/agent/services/multiSigService.ts

```diff
@@ -971,6 +971,64 @@ class MultiSigService extends AgentService {
       { gid: hab["prefix"] }
     );
   }
+
+  async multisigAdmit(
+    multisigSignifyName: string,
+    notificationSaid: string,
+    timeAdmit?: string
+  ) {
+    const exchangeMessage = await this.props.signifyClient
+      .exchanges()
+      .get(notificationSaid);
+    const grantSaid = exchangeMessage.exn.d;
+    const { ourIdentifier, multisigMembers } =
+      await this.getMultisigParticipants(multisigSignifyName);
+    const gHab = await this.props.signifyClient
+      .identifiers()
+      .get(multisigSignifyName);
+    const mHab = await this.props.signifyClient
+      .identifiers()
+      .get(ourIdentifier.signifyName);
+
+    const time = timeAdmit
+      ? timeAdmit
+      : new Date().toISOString().replace("Z", "000+00:00");
+
+    const [admit, sigs, end] = await this.props.signifyClient
+      .ipex()
+      .admit(multisigSignifyName, "", grantSaid, time);
```

Comment:
Hmm, this still isn't solving what I asked for during the week.

If this is the FIRST wallet, they do this by calling `ipex().admit()`.

But if this is the SECOND wallet joining, they should NOT be calling `ipex().admit()` even if it works. They should extract the IPEX message directly from the `/multisig/exn` message and re-sign it.

As I already said, for IPEX offer and grant messages - we can't simply just pass the timestamp, we need to extract the full message.

Reply:
I'm still not clear on how to handle this. Which method should we use to re-sign it?

Reply:
In Signify, there is
```
    async createExchangeMessage(
        sender: HabState,
        route: string,
        payload: Dict<any>,
        embeds: Dict<any>,
        recipient: string,
        datetime?: string,
        dig?: string
    ): Promise<[Serder, string[], string]> {
        const keeper = this.client.manager!.get(sender);
        const [exn, end] = exchange(
            route,
            payload,
            sender['prefix'],
            recipient,
            datetime,
            dig,
            undefined,
            embeds
        );

        const sigs = await keeper.sign(b(exn.raw));
        return [exn, sigs, d(end)];
    }
```

`exn` and `end` should be available. Can you check the embeds of the `/multisig/exn` and see if `end` is part of `atc`?

Because of:
```
    const gembeds = {
      exn: [admit, atc],
    };
```

Reply:
Hmm `atc` includes the sigers from the other person and the seal too. Lets just try it and see.

I'm pretty sure `end` is empty for admit but for others it won't be

Reply:
For SECOND, we extract the IPEX message directly from the `/multisig/exn` then we should call createExchangeMessage() with the route `/ipex/admit` or `/mutilsig/exn`. Or the flow is to use with route `/ipex/admit` first then `/multisig/exn`

Reply:
For SECOND, extract the IPEX message from the `/multisig/exn`. This gives you a `/ipex/admit`, which you can sign - you need to update `sigers` for creating `atc`.

Then you embed `admit` and the new `atc` inside `/multisig/exn` with `createExchangeMethod()`. Then you are done.

Reply:
All we are trying to do is to avoid rebuilding the admit message with `ipex().admit()` because for other messages it won't be so simple to re-build. (like grant, offer)